creator-node/src/services/stateMachineManager/stateReconciliation/recoverOrphanedData.jobProcessor.ts

```diff
@@ -18,40 +31,178 @@ const { QUEUE_NAMES } = require('../stateMachineConstants')
  * @param {Object} param.logger the logger that can be filtered by jobName and jobId
  */
 module.exports = async function ({
+  discoveryNodeEndpoint,
   logger
 }: DecoratedJobParams<RecoverOrphanedDataJobParams>): Promise<
   DecoratedJobReturnValue<RecoverOrphanedDataJobReturnValue>
 > {
-  // TODO: Clear redis state
-
-  // TODO: Query this node's db to find all users who have data on it and save them in a redis set
-  const usersOnThisNode = {}
-
-  /**
-   * TODO: Query Discovery to find all users who have this node as their primary or secondary.
-   *       This will do some kind of while loop that requests paginated responses from
-   *       `/v1/full/users/content_node/all?creator_node_endpoint=<THIS_CONTENT_NODE>` and adds
-   *       them to a redis set until reaching the end of the pagination
-   */
-  const usersWithThisNodeInReplica = {}
-
-  // TODO: Perform a set difference between usersOnThisNode and usersWithThisNodeInReplica.
-  //       https://redis.io/commands/sdiff/
-  const usersWithOrphanedData: any[] = []
-
-  /**
-   *  TODO: Run modified "force wipe" (modified forceResync=true flag) on each user with orphaned data:
-   *        1. Merge orphaned data from this node into the user's primary.
-   *        2. Wipe the user's data from this node.
-   *        3. *DON'T* resync from the primary to this node -- this is what non-modified forceResync would do.
-   * */
-
-  // Enqueue another job to search for any new data that gets orphaned after this job finishes
-  const jobsToEnqueue: JobsToEnqueue = {
-    [QUEUE_NAMES.RECOVER_ORPHANED_DATA]: [{}]
+  const numWalletsOnNode = await _storeWalletsOnThisNode()
+  const numUsersWithNodeInReplicaSet = await _storeWalletsWithThisNodeInReplica(
+    discoveryNodeEndpoint
+  )
+
+  // Wallets with orphaned data = set difference between walletsOnThisNode and walletsWithThisNodeInReplicaSet
+  const walletsWithOrphanedOrUnsyncedData: string[] = await redisClient.sdiff(
+    WALLETS_ON_NODE_KEY,
+    WALLETS_WITH_NODE_IN_REPLICA_SET_KEY
+  )
+
+  // TODO: Filter wallets that had this node in RS but didn't have data on this node (these are just unsynced users).
+  //       Could use SSCAN but that's O(N). Alternatively, when enqueuing the sync we could check which set the wallet is in and not wipe if the node is in the user's replica set
+  const walletsWithOrphanedData = walletsWithOrphanedOrUnsyncedData
+
+  // TODO: Filter wallets that were just checked in the previous job since their syncs won't have been processed yet?
```

Comment:
another detail for feedback @SidSethi @dmanjunath -- if one recover-orphaned-data job completes, it will enqueue a bunch of sync requests that will sync and then wipe for all the orphaned users. but the next recover-orphaned-data job will start running soon after and enqueue requests for the same users unless we have some kind of delay. do you think it's worth just saving the state in redis and ignoring wallets from the previous run or doing some kind of redis `expire`?

Reply:
synced on slack and decided to increase the rate limiter to once per 30 minutes or hour or something since no (or very few) new data will be orphaned between runs. this is also the simplest solution -- rather than worrying about TTLs we can just give the syncs some extra time to process before checking for more orphaned data